### ISSUE TITLE
Is this project still under development

### ISSUE THREAD

**Issue by xopt1x** [Netflix/bless#105]
After AWS released instance connect, I have to ask are you still working on this or should we switch over?

**Comment by patrickjahns** [comment]
AWS instance connect works quite different in terms of concepts - hence I would rephrase your question a little and ask: "how does bless differ from aws ec2-instance-connect"

AWS ec2-instance-connect consists of 2 aspects:
- [ec2 sendsshpublickey api](https://docs.aws.amazon.com/ec2-instance-connect/latest/APIReference/API_SendSSHPublicKey.html)
- [a tool that needs to be installed on the host](https://github.com/aws/aws-ec2-instance-connect-config)



First and foremost, lets have a quick look at the [ec2-instance-connect source](https://github.com/aws/aws-ec2-instance-connect-config) to check the implementation on the host side. At the core, instance-connect leverages the `AuthorizedKeysCommand` openssh feature:

On a connection attempt the executable specified via `AuthorizedKeysCommand` is invoked (with `$USER` as being the user that wants to connect ) to retrieve a public key.  In the case of instance-connect, the script will look at the instance metadata, specifically the url `ttp://169.254.169.254/latest/meta-data/managed-ssh-keys/active-keys/$USER`, and will retrieve the public key from there and login the respective user.

From my perspective, this is basically the same as login via a key written to the authorized keys file for a user. With the upside that the keys do not need to be pre-provisioned on the instance - but also still with the downsides of public keys

As previously mentioned, another puzzle piece for instance connect is the [SendSSHPublicKey api](https://docs.aws.amazon.com/ec2-instance-connect/latest/APIReference/API_SendSSHPublicKey.html), which is quite neat in the sense of we are able to manage access to it via IAM policies. However this is currently limited to two aspects of interest: limiting the instances one can sent the publickey to as well as the username on the node. 

On a side-note, currently aws instance-connect only supports RSA keys 



**BLESS** leverages ssh **certificates** which provides another set of capabilities natively built into openssh, such as "force-command" and "extensions" (which allow/deny specific features) - more on that can be found via [openssh documentation](https://cvsweb.openbsd.org/src/usr.bin/ssh/PROTOCOL.certkeys?annotate=HEAD)

There's been quite a few blog posts about bless ( and ssh certificates in general ) so I won't go into too much detail here.



From my point of view, it boils down to the difference between public/priv key and certificates with its pros/cons for both approaches. AWS instance-connect provides a cherry on-top of the normal hassle with public/privatekeys, as it allows for easy (temporary) deployment of keys that can be controlled with IAM policies. Utilizing certificates (and thus using bless) provides more control for permitted actions on the host node itself and adds another layer of control. 


Note: I know it does not answer the question initial question if its supported - but I felt like sharing here my personal view on the matter as I surely hope that bless continues to be supported (as open source) 😄

**Comment by xopt1x** [comment]
^ mansplaining 10x.

I will assume because Netflix is not answering that this project is probably deprecated and we should unroll it from our organization.

**Comment by le0pard** [comment]
@xopt1x you can try similar on golang: https://github.com/certonid/certonid (it also contain CLI)
